runtime/queries/metricsview_timeseries.go

```diff
@@ -235,10 +159,26 @@ func (q *MetricsViewTimeSeries) resolveDruid(ctx context.Context, olap drivers.O
 			return err
 		}
 
+		if nullRecords == nil {
+			nullRecords = generateNullRecords(records)
+		}
+		if zeroTime.Equal(start) {
+			if q.TimeStart != nil {
+				start = truncateTime(q.TimeStart.AsTime(), q.TimeGranularity, tz, int(fdow), int(fmoy))
+				data = addNulls(data, nullRecords, start, t, q.TimeGranularity)
+			}
+		} else {
+			data = addNulls(data, nullRecords, start, t, q.TimeGranularity)
+		}
+
 		data = append(data, &runtimev1.TimeSeriesValue{
 			Ts:      timestamppb.New(t),
 			Records: records,
 		})
+		start = addTo(t, q.TimeGranularity)
+	}
+	if q.TimeEnd != nil && nullRecords != nil {
```

Comment:
If we move init of `nullRecords` outside of the for loop then we can remove the `nullRecords != nil` condition as well.